tests/async_tests/test_interfaces.py

```diff
@@ -212,6 +212,25 @@ async def test_http_proxy(proxy_server: URL, proxy_mode: str) -> None:
         assert reason == b"OK"
 
 
+@pytest.mark.parametrize("local_address", [b"0.0.0.0"])
+@pytest.mark.asyncio
+# This doesn't run with trio, since trio doesn't support local_address.
+async def test_http_request_local_address(local_address: str) -> None:
+    async with httpcore.AsyncConnectionPool(local_address=local_address) as http:
+        method = b"GET"
+        url = (b"http", b"example.org", 80, b"/")
+        headers = [(b"host", b"example.org")]
+        http_version, status_code, reason, headers, stream = await http.request(
+            method, url, headers
+        )
+        body = await read_body(stream)
+
+        assert http_version == b"HTTP/1.1"
+        assert status_code == 200
+        assert reason == b"OK"
+        assert len(http._connections[url[:3]]) == 1  # type: ignore
```

Comment:
It doesn't look like this test tests that the `local_address` actually gets used — I suppose that's intended for simplicity since testing that w/o mocking is not trivial?